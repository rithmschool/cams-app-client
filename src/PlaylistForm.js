import React, {PropTypes, Component} from 'react';
import axios from 'axios';

const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'
let userId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

class PlaylistForm extends Component {

  constructor(props){
    super(props)
    this.state = {
      playlistTitle: "",
      videoUrl: ""
    }
  }

  static contextTypes = {
        router: PropTypes.object
    }

  addPlaylist(config, thisArg) {
    axios.post(`${BASE_URL}/api/users/${userId}/playlists`, thisArg.state, config).then(response => {
      localStorage.setItem('token',response.data.token)
      thisArg.context.router.history.push('/dashboard')
    })
  }

  handleChange(e){
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit(event){
      let config = {
        headers: {
          'Accept':'application/json',
          'ContentType':'application/json',
          'Authorization': 'bearer ' + localStorage.getItem('token')
        }
      }
      event.preventDefault()
      this.addPlaylist(config, this)
  }

  render(){
    return(
      <form onSubmit={this.handleSubmit.bind(this)}>
        <input type="string" name="playlistTitle" placeholder="Playlist Name"/>
        <button type="submit" value="Submit">Submit</button>
      </form>
    )
  }
}

export default PlaylistForm;
