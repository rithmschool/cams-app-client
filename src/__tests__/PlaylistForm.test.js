import PlaylistForm from "../PlaylistForm";
import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";

test("matches snapshot of PlaylistForm", () => {
  const playlistForm = renderer
    .create(<PlaylistForm error={false} addPlaylist={function() {}} />)
    .toJSON();
  expect(playlistForm).toMatchSnapshot();
});
