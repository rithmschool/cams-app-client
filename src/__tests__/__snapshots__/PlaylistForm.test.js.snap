// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of PlaylistForm 1`] = `
<div>
  <h3
    className=""
  >
    Add Playlist
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Playlist Name"
      required={true}
      type="string"
      value=""
    />
    <button
      className="button button-hover"
      type="submit"
      value="Submit"
    >
      Submit
    </button>
  </form>
</div>
`;
